src/nvim/window.c

```diff
@@ -2973,6 +2973,7 @@ int win_new_tabpage(int after)
     return FAIL;
   }
   curtab = newtp;
+  apply_autocmds(EVENT_TABNEW, arg, arg, false, curbuf);
 
   /* Create a new empty window. */
   if (win_alloc_firstwin(tp->tp_curwin) == OK) {
```

Comment:
Yes, that seems better to me. After all, the success condition of `win_new_tabpage` only happens within that branch, it would be weird to trigger the event when the function fails (not sure how likely that is though).